test/catalog/catalog_test.cpp

```diff
@@ -303,5 +304,105 @@ TEST_F(CatalogTests, DroppingCatalog) {
   EXPECT_NE(nullptr, catalog);
 }
 
+TEST_F(CatalogTests, LayoutCatalogTest) {
+  // This test creates a table, changes its layout.
+  // Create another additional layout.
+  // Ensure that default is not changed.
+  // Drops layout and verifies that the default_layout is reset.
+  // It also queries pg_layout to ensure that the entry is removed.
+
+  auto db_name = "temp_db";
+  auto table_name = "temp_table";
+  auto catalog = catalog::Catalog::GetInstance();
+  // Create database.
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog->CreateDatabase(db_name, txn);
+
+  // Create table.
+  auto val0 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val0", true);
+  auto val1 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val1", true);
+  auto val2 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val2", true);
+  auto val3 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val3", true);
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({val0, val1, val2, val3}));
+  catalog->CreateTable(db_name, DEFAULT_SCHEMA_NAME, table_name,
+                       std::move(table_schema), txn);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  auto table =
```

Comment:
Try to use **table catalog object** instead of **data table** to obtain information like database oid and table oid.